engine/jobmaster/dm/utils.go

```diff
@@ -19,5 +19,5 @@ import (
 
 // NewDMResourceID returns a ResourceID in DM's style. Currently only support local resource.
 func NewDMResourceID(taskName, sourceName string) resModel.ResourceID {
-	return "/" + string(resModel.ResourceTypeLocalFile) + "/" + taskName + "/" + sourceName
+	return "/" + string(resModel.ResourceTypeS3) + "/" + taskName + "-" + sourceName
```

Comment:
Is this a definite or configurable option? If it's definite, what happens if the executor is not configured with s3?

Reply:
The current implementation is deterministic, which would cause errors when s3 is not configured. To solve this problem, we need to figure out: 
1. Add a configurable parameter for the dm job, or get the information about whether the framework is configured with s3 through the BaseWorker interface?
2. Restore the failover logic of the dm so that all dump jobs are created with resource Constraint.

Reply:
cc @amyangfei @lance6716

Reply:
Maybe we could fix this in another pr.

Reply:
What about exposing an option to decide using s3 or local file in DM job config. 
- Maybe in the future this is decided by user or dataflow service
- Besides adding an option makes it easy to debug in local environment, without of setup a minio server.

Reply:
This feature will be implemented in other pr.